app/models/concerns/orchestration/dns.rb

```diff
@@ -1,175 +1,211 @@
 
+  RECORD_TYPES = [:a, :aaaa, :ptr, :ptr6]
+
   included do
     after_validation :dns_conflict_detected?, :queue_dns, :unless => :importing_facts
     before_destroy :queue_dns_destroy, :unless => :importing_facts
     register_rebuild(:rebuild_dns, N_('DNS'))
   end
 
-  def dns?
+  def dns_ready?
     # host.managed? and managed? should always come first so that orchestration doesn't
     # even get tested for such objects
-    (host.nil? || host.managed?) && managed? && hostname.present? && ip_available? && !domain.nil? && !domain.proxy.nil? && SETTINGS[:unattended]
+    SETTINGS[:unattended] && (host.nil? || host.managed?) && managed? && hostname.present?
+  end
+
+  def dns?
+    dns_ready? && ip_available? && !domain.nil? && !domain.proxy.nil?
+  end
+
+  def dns6?
+    dns_ready? && ip6_available? && !domain.nil? && !domain.proxy.nil?
   end
 
   def reverse_dns?
-    # host.managed? and managed? should always come first so that orchestration doesn't
-    # even get tested for such objects
-    (host.nil? || host.managed?) && managed? && hostname.present? && ip_available? && !subnet.nil? && subnet.dns? && SETTINGS[:unattended]
+    dns_ready? && ip_available? && !subnet.nil? && subnet.dns?
+  end
+
+  def reverse_dns6?
+    dns_ready? && ip6_available? && !subnet6.nil? && subnet6.dns?
   end
 
   def rebuild_dns
-    logger.info "DNS not supported for #{name}, #{ip}, skipping orchestration rebuild" unless dns?
-    logger.info "Reverse DNS not supported for #{name}, #{ip}, skipping orchestration rebuild" unless reverse_dns?
-    return true unless dns? || reverse_dns?
+    logger.info "IPv4 DNS not supported for #{name}, skipping orchestration rebuild" unless dns?
+    logger.info "IPv6 DNS not supported for #{name}, skipping orchestration rebuild" unless dns6?
+    logger.info "Reverse IPv4 DNS not supported for #{name}, skipping orchestration rebuild" unless reverse_dns?
+    logger.info "Reverse IPv6 DNS not supported for #{name}, skipping orchestration rebuild" unless reverse_dns6?
+    return true unless dns? || dns6? || reverse_dns? || reverse_dns6?
     del_dns_a_record_safe
+    del_dns_aaaa_record_safe
     del_dns_ptr_record_safe
-    a_record_result, ptr_record_result = true, true
+    del_dns_ptr6_record_safe
+    a_record_result = true
+    aaaa_record_result = true
+    ptr_record_result = true
+    ptr6_record_result = true
     begin
       a_record_result = recreate_a_record if dns?
+      aaaa_record_result = recreate_aaaa_record if dns6?
       ptr_record_result = recreate_ptr_record if reverse_dns?
-      a_record_result && ptr_record_result
+      ptr6_record_result = recreate_ptr6_record if reverse_dns6?
+      a_record_result && aaaa_record_result && ptr_record_result && ptr6_record_result
     rescue => e
-      Foreman::Logging.exception "Failed to rebuild DNS record for #{name}, #{ip}", e, :level => :error
+      Foreman::Logging.exception "Failed to rebuild DNS record for #{name}", e, :level => :error
```

Comment:
I think logging the IP of the interface (or ip6) is worth it here, so you know which interface is failing more easily